src/main/scala/is/hail/io/RowStore.scala

```diff
@@ -592,16 +592,16 @@ final class PackDecoder(in: InputBuffer) extends Decoder {
     aoff
   }
 
-  def readStruct(t: TStruct, region: Region, offset: Long) {
+  def readStructBase(t: TStructBase, region: Region, offset: Long) {
     val nMissingBytes = t.nMissingBytes
     in.readBytes(region, offset, nMissingBytes)
 
     var i = 0
     while (i < t.size) {
       if (t.isFieldDefined(region, offset, i)) {
         val off = offset + t.byteOffsets(i)
-        t.fieldType(i) match {
-          case t2: TStruct => readStruct(t2, region, off)
+        t.types(i) match {
```

Comment:
I want you to benchmark read performance on a reasonably sized table (ExAC metadata, maybe). I'm guessing it's ~3x slower because of this tiny change!